contracts/services/astroport-lper/src/astroport_cw20.rs

```diff
@@ -85,7 +85,8 @@ fn create_xyk_liquidity_msg(
     let halved_coin = cosmwasm_std::Coin {
         denom: asset_balance.denom.clone(),
         amount: cosmwasm_std::Uint128::from(asset_balance.amount.u128())
-            / cosmwasm_std::Uint128::from(2u128),
+            .checked_div(cosmwasm_std::Uint128::from(2u128))
+            .expect("denominator is not zero; qed"),
```

Comment:
does this make sense when the denominator is hardcoded to 2? also can't we use `?` to unpack the result of division?

Reply:
It's a common way of making more explicit invariants that you know to be true / infaillible (I've seen this used in many codebases) i.e. it's clearer to use an `.expect` with a message than just an `.unwrap`.

Reply:
Only thing a `checked_div` checks here is for division by zero, and we are dividing by a hardcoded value (2) so I'm not sure if it's any cleaner. Imho it's just extra operations.

Regarding `.expect()`, totally agree that we shouldn't simply `.unwrap()`. Personally I see `.expect()` as very similar to unwrap in that it still panics, just with a custom message. In contrast with `?`, which propagates the underlying error message. I think that propagating underlying errors, especially for stuff like cosmwasm math, is the correct approach because we would always receive the same uniform error messages across our codebase.
Another point is regarding consistency, - use `?` much more than `.expect()`.

Regardless, my approval is already here as the way it's currently done would (ofc) work. Just nitpicking about uniformity.